src/nrf5_sdk15_platform/communication/ruuvi_nrf5_sdk15_communication_ble_advertising.c

```diff
@@ -651,10 +651,19 @@ rd_status_t ri_adv_scan_start (const uint32_t window_interval_ms,
     scan_params.active = 0; // Do not scan for scan responses
     ruuvi_nrf5_sdk15_radio_channels_set (scan_params.channel_mask, m_radio_channels);
 
+    if (RUUVI_NRF5_SDK15_ADV_EXTENDED_ENABLED)
+    {
+        scan_params.extended = 1;
+    }
+    else
+    {
+        scan_params.extended = 0;
+    }
+
     // Other than 1 MBit / s require extended advertising.
     if (BLE_GAP_PHY_1MBPS == scan_phys)
     {
-        scan_params.extended = 0;
+        scan_params.extended = 1;
```

Comment:
Was this tested on hardware? I was under impression that 2 MBit/s extended advertisements send a primary advertisement at 1 MBit / s and primary advertisement contains information about when and where the actual extended payload will be.